sdk/core/ts-http-runtime/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -130,6 +136,47 @@ function getChallenge(response: PipelineResponse): string | undefined {
   return;
 }
 
+/**
+ * Checks if the request is allowed to be sent over an insecure connection.
+ *
+ * A request is allowed to be sent over an insecure connection when:
+ * - The `allowInsecureConnection` option is set to `true`.
+ * - The request has the `allowInsecureConnection` property set to `true`.
+ * - The request is being sent to `localhost` or `127.0.0.1`
+ */
+function allowInsecureConnection(
+  request: PipelineRequest,
+  options: BearerTokenAuthenticationPolicyOptions,
+): boolean {
+  if (options.allowInsecureConnection && request.allowInsecureConnection) {
```

Comment:
I think I prefer using an environment variable for this since it eliminates changing the client options between Azure and unbranded core and also is less likely to be accidentally checked in